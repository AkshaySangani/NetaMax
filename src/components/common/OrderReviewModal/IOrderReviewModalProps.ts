import { IProductDetail } from "dataflows/MisCompras/IProductDetail";

export interface IOrderReviewModalProps {
  /**
   * The products order.
   * @type {Pick<IProductDetail, "quantity" | "id" | "seoFilename" | "name" | "price">[]}
   */
  productsOrder: Pick<IProductDetail, "quantity" | "id" | "seoFilename" | "name" | "price">[];

  /**
   * The subtotal amount of the order.
   * @type {number}
   */
  orderSubtotalInclTax: number;

  /**
   * indicates if the modal is open.
   * @type {boolean}
   */
  isOpen: boolean;

  /**
   * The close modal function.
   * @type {() => void}
   */
  onClose: () => void;
}
