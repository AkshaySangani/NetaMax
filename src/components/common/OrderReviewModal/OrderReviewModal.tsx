import React, { ReactElement } from "react";

import { UNITS_TITLE } from "constants/basketConstants";
import { CURRENCY_NAME } from "constants/localeFormatConstants";
import { screenSizes } from "styled/screen";
import { formatMoney } from "utils/currencyUtils";

import {
  useMediaQuery,
  Box,
  Center,
  Divider,
  Flex,
  HStack,
  Img,
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  Stack,
  Text,
} from "@chakra-ui/react";

import { IOrderReviewModalProps } from "./IOrderReviewModalProps";

/**
 *  The order review modal component
 * @param {IOrderReviewModalProps} props the component props
 * @returns {ReactElement} The order review modal component
 */
export const OrderReviewModal = ({
  productsOrder,
  orderSubtotalInclTax,
  isOpen,
  onClose,
}: IOrderReviewModalProps): ReactElement => {
  const [isDesktop] = useMediaQuery(`(min-width: ${screenSizes.mdMin}px)`);

  return (
    <Center>
      <Modal
        isOpen={isOpen}
        onClose={onClose}
        size={isDesktop ? "4xl" : "sm"}
        scrollBehavior="inside"
      >
        <ModalOverlay />
        <ModalContent>
          <ModalCloseButton fontSize="18px" />
          <ModalHeader w="100%" justifyContent="flex-start">
            <Text fontSize="21px" textAlign="left" marginTop="10px">
              Tus Productos
            </Text>
          </ModalHeader>
          <ModalBody w="100%" maxHeight="550px">
            {productsOrder &&
              productsOrder.map((item) => (
                <Box key={item.id}>
                  <Stack marginBottom="12px" alignItems="center" direction="row">
                    <Box display="flex" alignItems="center" justifyContent="center" w="20%">
                      <Img
                        rounded="lg"
                        fit="cover"
                        height="45px"
                        src={item.seoFilename}
                        alt={item.name}
                        draggable="false"
                        loading="lazy"
                      />
                    </Box>
                    <Box w="77%" pt="4">
                      <Stack spacing="0.5">
                        <Text
                          isTruncated
                          maxWidth="86%"
                          fontWeight="700"
                          fontSize={{ base: "sm", md: "sm", xl: "18px" }}
                        >
                          {item.name}
                        </Text>

                        <Flex alignItems="center">
                          <Text fontWeight="700" fontSize={{ base: "sm", md: "sm", xl: "16px" }}>
                            Precio unitario:
                          </Text>
                          <Text ml="4px" fontWeight="500">
                            {formatMoney(parseFloat(item.price), CURRENCY_NAME)}
                          </Text>
                        </Flex>

                        <HStack spacing="1" mt="3" justify="space-between">
                          <Text fontWeight="700" fontSize={{ base: "sm", md: "sm", xl: "16px" }}>
                            {UNITS_TITLE}: {item.quantity}
                          </Text>
                        </HStack>
                      </Stack>
                    </Box>
                  </Stack>
                  <Divider />
                </Box>
              ))}
          </ModalBody>
          <ModalFooter
            w="100%"
            h="53px"
            justifyContent="space-between"
            borderRadius="12px"
            bg="#F0F0F0"
            fontWeight="700"
            boxShadow="0px 0px 7px rgba(0, 0, 0, 0.08)"
            position="relative"
          >
            <Text>Total</Text>
            <Text>{formatMoney(Number(orderSubtotalInclTax), CURRENCY_NAME)}</Text>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </Center>
  );
};
