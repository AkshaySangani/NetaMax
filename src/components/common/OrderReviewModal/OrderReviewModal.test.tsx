import { render, cleanup } from "@testing-library/react";
import { OrderReviewModal } from "./OrderReviewModal";
import { IOrderReviewModalProps } from "./IOrderReviewModalProps";

describe("OrderReviewModal component", () => {
  let mock: IOrderReviewModalProps;

  beforeEach(() => {
    mock = {
      productsOrder: [
        {
          name: "Cigarros Pall Mall Mykonos",
          id: "460",
          quantity: "1",
          price: "75",
          seoFilename: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002615_0.jpeg",
        },
        {
          name: "Pasta Dental Colgate Triple Accion Original 50 M",
          id: "467",
          quantity: "1",
          price: "75",
          seoFilename: "https://nopcommerce-assets.s3.us-east-2.amazonaws.com/0002629_0.jpeg",
        },
      ],
      orderSubtotalInclTax: 150.78,
      isOpen: false,
      onClose: jest.fn(),
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("OrderReviewModal should be render correctly", () => {
    const { container } = render(<OrderReviewModal {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
