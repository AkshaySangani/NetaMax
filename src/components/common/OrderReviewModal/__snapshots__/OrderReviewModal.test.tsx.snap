// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderReviewModal component OrderReviewModal should be render correctly 1`] = `
<div>
  <div
    class="css-gmuwbf"
  />
</div>
`;
